pkg/agent/clean/credentials.go

```diff
@@ -0,0 +1,68 @@
+package clean
+
+import (
+	"context"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// UnusedCattleCredentials removes all of the cattle-credentials that aren't being used by the current pod on a ticker. meant to be run from a goroutine so it doesn't stop the parent execution
+func UnusedCattleCredentials() {
+	for range time.Tick(time.Minute) {
```

Comment:
Don't need to run every minute, new secrets are created only when token/url/namespace changes